### PR TITLE
StakingPage: Support sorting by validator name

### PR DESCRIPTION
**Pull request by matevz**
Fixes sorting by name on the validator list page. If validator name is not present, sort by oasis address.

image

### CODE REVIEW

File: src/app/pages/StakingPage/Features/ValidatorList/index.tsx [hunk 1]
```diff
@@ -74,6 +75,11 @@ export const ValidatorList = memo((props: Props) => {
             <ShortAddress address={datum.address} />
           </Text>
         ),
+      sortable: true,
+      sortFunction: (row1, row2) =>
+        row1.name
+          ? row1.name.localeCompare(row2.name ?? row2.address)
+          : row1.address.localeCompare(row2.name ?? row2.address),
```
Comment (lukaw3d) [inline src/app/pages/StakingPage/Features/ValidatorList/index.tsx:82]:
```suggestion
      sortFunction: (row1, row2) => (row1.name ?? row1.address).localeCompare(row2.name ?? row2.address),
```
is more symmetric

**Review by lukaw3d — APPROVED** [review]

File: src/app/pages/StakingPage/Features/ValidatorList/index.tsx (hunk 1 above)
Comment (pro-wh) [reply src/app/pages/StakingPage/Features/ValidatorList/index.tsx:82]: agreed with this suggestion above

could you also confirm that `??` is appropriate? surrounding code, e.g. `datum.name ? ( ... ) : ( ... )`, suggests that `&&` would be consistent

Comment (matevz) [reply src/app/pages/StakingPage/Features/ValidatorList/index.tsx]: Thanks, I applied the suggested change and also used `??` for `datum.name ? ( ... ) : ( ... )` above.

**Comment by matevz** [comment]
I noticed sorting by name also didn't work in the Active delegations and Debonding delegations pages as well as sorting by Amount. Fixed those as well.

**Pull request merged**